src/main/java/de/tum/in/www1/artemis/service/notifications/MailService.java

```diff
@@ -168,14 +163,13 @@ public void sendSAML2SetPasswordMail(User user) {
     /**
      * Sets the context and subject for the case that the notificationSubject is a Post
      *
-     * @param context             that is modified
      * @param notificationSubject which has to be a Post
      * @param locale              used for translations
      * @return the modified subject of the email
      */
-    private String setPostContextAndSubject(Context context, Object notificationSubject, Locale locale) {
-        // For Announcement Posts
-        String newAnnouncementString = locale.toString().equals("en") ? newAnnouncementEN : newAnnouncementDE;
+    private String createAnnouncementText(Object notificationSubject, Locale locale) {
+        // Translation that can not be done via i18n Resource Bundle (for Thymeleaf) but has to be set in this service via Java
+        String newAnnouncementString = locale.toString().equals("en") ? "New announcement \"%s\" in course \"%s\"" : "Neue Ankündigung \"%s\" im Kurs \"%s\"";
```

Comment:
```suggestion
        String newAnnouncementString = "en".equals(locale.toString()) ? "New announcement \"%s\" in course \"%s\"" : "Neue Ankündigung \"%s\" im Kurs \"%s\"";
```
Codacy Static Code Analysis warning